protocols/raft/src/main/java/io/atomix/protocols/raft/roles/FollowerRole.java

```diff
@@ -85,14 +86,15 @@ public FollowerRole(RaftContext context) {
   @Override
   public synchronized CompletableFuture<Void> stop() {
     raft.getMembershipService().removeListener(clusterListener);
-    return super.stop().thenRun(this::cancelHeartbeatTimers);
+    return super.stop().thenRun(this::cancelHeartbeatTimer);
   }
 
   /** Cancels the heartbeat timer. */
-  private void cancelHeartbeatTimers() {
-    if (heartbeatTimoutTimer != null) {
+  private void cancelHeartbeatTimer() {
+    if (heartbeatTimer != null) {
```

Comment:
this can also be called from different threads right?

Reply:
Hm, the heartbeat thread is closed only when the partition is closed, right? So it should be safe to schedule it the close there and not wait for it, I guess. :+1: